src/App.js

```diff
@@ -35,6 +39,14 @@ function App() {
       return <Navigate to="/login" />;
     }
   }
+  function AdminRouteMiddleware({ children }) {
+    const { isAdmin, checkAdminAuth } = useAuth();
```

Comment:
No need to import the isAdmin flag, as the checkAdminAuth function is already returning true/false based on the whether admin is logged in or not.
const adminLoggedIn = checkAdminAuth();
Based on adminLoggedIn, return the children or navigate to admin-login page